File: samples/server/petstore/scala-play-server/app/api/StoreApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.Order


@Singleton
class StoreApiController @Inject()(cc: ControllerComponents, api: StoreApi) extends AbstractController(cc) {
  /**
    * DELETE /v2/store/order/:orderId
    * @param orderId ID of the order that needs to be deleted
    */
  def deleteOrder(orderId: String): Action[AnyContent] = Action { request =>
    def executeApi(): Unit = {
      api.deleteOrder(orderId)
    }

    executeApi()
    Ok
  }

  /**
    * GET /v2/store/inventory
    */
  def getInventory(): Action[AnyContent] = Action { request =>
    def executeApi(): Map[String, Int] = {
      api.getInventory()
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /v2/store/order/:orderId
    * @param orderId ID of pet that needs to be fetched
    */
  def getOrderById(orderId: Long): Action[AnyContent] = Action { request =>
    def executeApi(): Order = {
      api.getOrderById(orderId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /v2/store/order
    */
  def placeOrder(): Action[AnyContent] = Action { request =>
    def executeApi(): Order = {
      val order = request.body.asJson.map(_.as[Order]).getOrElse {
        throw new OpenApiExceptions.MissingRequiredParameterException("body", "order")
      }
      api.placeOrder(order)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: samples/server/petstore/typescript-nestjs-server/builds/default/controllers/StoreApi.controller.ts
import { Body, Controller, Delete, Get, Post, Param, Query, Req } from '@nestjs/common';
import { Observable } from 'rxjs';
import { StoreApi } from '../api';
import { Order,  } from '../models';

@Controller()
export class StoreApiController {
  constructor(private readonly storeApi: StoreApi) {}

  @Delete('/store/order/:orderId')
  deleteOrder(@Param('orderId') orderId: string, @Req() request: Request): void | Promise<void> | Observable<void> {
    return this.storeApi.deleteOrder(orderId, request);
  }

  @Get('/store/inventory')
  getInventory(@Req() request: Request): { [key: string]: number; } | Promise<{ [key: string]: number; }> | Observable<{ [key: string]: number; }> {
    return this.storeApi.getInventory(request);
  }

  @Get('/store/order/:orderId')
  getOrderById(@Param('orderId') orderId: number, @Req() request: Request): Order | Promise<Order> | Observable<Order> {
    return this.storeApi.getOrderById(orderId, request);
  }

  @Post('/store/order')
  placeOrder(@Body() order: Order, @Req() request: Request): Order | Promise<Order> | Observable<Order> {
    return this.storeApi.placeOrder(order, request);
  }

} 